cognite/client/utils/_concurrency.py

```diff
@@ -109,38 +110,109 @@ def collect_exc_info_and_raise(
         ) from dup_exc
 
 
-def execute_tasks_concurrently(
-    func: Callable, tasks: Union[Sequence[Tuple], List[Dict]], max_workers: int
-) -> TasksSummary:
+T_Result = TypeVar("T_Result", covariant=True)
+
+
+class Executor(Protocol):
+    def submit(self, fn: Callable[..., T_Result], *args: Any, **kwargs: Any) -> Future[T_Result]:
+        ...
+
+
+class Future(Protocol[T_Result]):
+    def result(self) -> T_Result:
+        ...
+
+
+class SyncFuture(Future):
+    def __init__(self, fn: Callable[..., T_Result], *args: Any, **kwargs: Any):
+        self.__fn = fn
+        self.__args = args
+        self.__kwargs = kwargs
+
+    def result(self) -> T_Result:
+        return self.__fn(*self.__args, **self.__kwargs)
+
+
+class MainThreadExecutor(Executor):
+    """
+    In order to support executing sdk methods in the browser using pyodide (a port of CPython to webassembly),
+    we need to be able to turn off the usage of threading. So we have this executor which implements the Executor
+    protocol but just executes everything serially in the main thread.
+    """
+
+    def submit(self, fn: Callable, *args: Any, **kwargs: Any) -> SyncFuture:
+        return SyncFuture(fn, *args, **kwargs)
+
+
+_THREAD_POOL_EXECUTOR_SINGLETON: ThreadPoolExecutor
+_MAIN_THREAD_EXECUTOR_SINGLETON = MainThreadExecutor()
+
+
+def _shutdown_thread_pool_executor() -> None:
+    try:
+        tpe = _THREAD_POOL_EXECUTOR_SINGLETON
+        tpe.shutdown(wait=False)
+    except NameError:
+        # TPE has not been initialized
+        pass
+
+
+atexit.register(_shutdown_thread_pool_executor)
+
+
+class Settings:
+    executor_type: Literal["threadpool", "mainthread"] = "threadpool"
```

Comment:
Maybe `ExecutorSettings` or something a bit more descriptive?  🤔 

Is the plan that we import this and set `Settings.executor_type = "mainthread"` when running with pyodide?

Reply:
yes and yes.